scripts/bundle.mjs

```diff
@@ -0,0 +1,134 @@
+import esbuild from 'esbuild';
+import { readFileSync, writeFileSync } from "fs";
+import { join, relative } from "path";
+import { cwd } from "./utils.mjs";
+const CWD = cwd(import.meta.url);
+const ROOT = join(CWD, '..');
+const SHOW_METAFILE = process.argv.some(string => string === '--metafile');
+
+/**
+ * Deliberately not try-catching around these operations.
+ * Node's errors around paths etc. are descriptive and anything that
+ * fails here should fail the process.
+ */
+(async () => {
+    // default build
+    await bundle({});
+
+    // debug build
+    await bundle({
+        plugins: [],
+        outfile: join(ROOT, 'dist/autofill-debug.js')
+    });
+})();
+
+/**
+ * @param {import("esbuild").BuildOptions} buildOptions
+ */
+async function bundle(buildOptions = {}) {
+    /** @type {import("esbuild").BuildOptions} */
+    const config = {
+        entryPoints: [join(ROOT, 'src/autofill.js')],
+        target: 'es2021',
+        bundle: true,
+        format: 'iife',
+        legalComments: 'inline',
+        outfile: join(ROOT, 'dist/autofill.js'),
+        metafile: true,
+        write: false,
+        loader: {
+            // import css files as text
+            '.css': 'text'
+        },
+        plugins: [zodReplacerPlugin()],
+        ...buildOptions
+    };
+
+    const buildOutput = await esbuild.build(config);
+
+    //
+    const files = buildOutput.outputFiles?.map(outputFile => {
+        const content = Buffer.from(outputFile.contents).toString('utf-8')
+        return {
+            original: content,
+            /**
+             * NOTE: This is here because esbuild strips comments - but it doesn't
+             * strip 'legal' comments. In the build output there will be instances of
+             *
+             * //! INJECT foobar here
+             *
+             * and we replace it to be this instead:
+             *
+             * // INJECT foobar here
+             *
+             */
+            replaced: content.replace(/\/\/! INJECT/g, '// INJECT'),
+            path: outputFile.path
+        }
+    })
+
+    for (let file of files||[]) {
+        writeFileSync(file.path, file.replaced)
+        console.log('✅', relative(ROOT, file.path))
+    }
+
+    if (SHOW_METAFILE && 'metafile' in buildOutput && buildOutput.metafile) {
+        console.log(await esbuild.analyzeMetafile(buildOutput.metafile))
+    }
+}
+
+/**
+ * This plugin is used to replace all the `export const` lines
+ * in the validators with `export const x = null;` This is done
+ * to strip `zod` from the final build.
+ */
+function zodReplacerPlugin() {
```

Comment:
this, and the function below, were just extracted from separate files/scripts. They belong together, so I got rid of the extra files